examples/systemcalls/systemcalls.c

```diff
@@ -16,8 +36,40 @@ bool do_system(const char *cmd)
  *   and return a boolean true if the system() call completed with success
  *   or false() if it returned a failure
 */
+    int ret = system(cmd);
+    //if system call fails, return error.
+    if(ret == -1)
+    {
+        perror("System error");
+        return false;
+    }
+    //if command is NULL, if system returns non-zero then shell is available, 0 means shell is not available
+    if(cmd == NULL)
+    {
+        if(!ret)
+        {
+            printf("No Shell Available");
+            return false;
+        }
+        else
+        {
+            printf("Shell Available");
+            return true;
+        }
+    }
+    // if cmd returns non zero value then return false, else true
+    else
```

Comment:
@GuruprashanthKrishnakumar I think you could simplify the logic here with `WIFEXITED` and `WEXITSTATUS` macros